### ISSUE TITLE
Any plans for Advisory Lock support?

### ISSUE THREAD

**Issue by klym1** [DbUp/DbUp#392]
As it was mentioned in https://github.com/DbUp/DbUp/issues/323 there are some DBs that do not support DDL modifications/rollback in a transaction (MySql in particular). But this option would be very helpful in a plenty of scenarios.  E.g. golang-migrate provides such support by leveraging  `GET_LOCK()/RELEASE_LOCK()` Api:
https://github.com/golang-migrate/migrate/tree/master/database/mysql

**Comment by droyad** [comment]
Would this be best done by the consumer themselves to suit their logic? Perhaps we just add an example of how to wrap DB up inside this kind of lock to the documentation.

**Comment by klym1** [comment]
Of course, consumers can do that by themselves. And they often do that. They even write various migration tools by themselves. But since there's already DbUp - why not make it more convenient and robust for the users?

**Comment by firedog** [comment]
Sample implementation of using advisory locks for SQL Server can be found in #358
